OpenSim/Common/Component.h

```diff
@@ -2180,18 +2184,21 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     * another Component. It serves as a placeholder for the Component and its
     * type and enables the Component to automatically traverse its dependencies
     * and provide a meaningful message if the provided Component is
-    * incompatible or non-existant.
+    * incompatible or non-existant. The connecteeNameIndex is provided to this 
+    * function via the OpenSim_DECLARE_CONNECTOR macro, and identifies the
+    * property in this Component that holds the connectee names for this
+    * connector.
     */
     template <typename T>
-    int constructConnector(const std::string& name, bool isList = false) {
-        int ix = updProperty_connectors().adoptAndAppendValue(
-            new Connector<T>(name, SimTK::Stage::Topology, *this));
-        // Add pointer to connectorsTable so we can access connectors easily by
-        // name.
-        _connectorsTable[name] = ix;
-        return ix;
+    bool constructConnector(const std::string& name,
+                            const PropertyIndex& connecteeNameIndex) {
```

Comment:
Great! I think calling macro generated methods in custom copy would be fine and we already have a precedent for that with `copyProperty_property_name` so I think it would at least be consistent.